Master is broken at build-time when SRAM is mapped at an high address

This is a regression from v2.5.0.

When trying to compile `hello_world` using `CONFIG_SRAM_BASE_ADDRESS=0x200880000` the compilation fails with:
```
[1/9] Linking C executable zephyr/zephyr_prebuilt.elf
FAILED: zephyr/zephyr_prebuilt.elf zephyr/zephyr_prebuilt.map 
: && ccache /home/carlo/work/proj/intel/sdk/aarch64-zephyr-elf/bin/aarch64-zephyr-elf-gcc    zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj  -o zephyr/zephyr_prebuilt.elf  -Wl,-T  zephyr/linker_zephyr_prebuilt.cmd  -Wl,-Map=/home/carlo/work/proj/intel/zephyr/zephyr-master/build/zephyr/zephyr_prebuilt.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/aarch64/libarch__arm__core__aarch64.a  zephyr/arch/arch/arm/core/aarch64/mmu/libarch__arm__core__aarch64__mmu.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/lib/posix/liblib__posix.a  zephyr/drivers/serial/libdrivers__serial.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj  -L"/home/carlo/work/proj/intel/sdk/aarch64-zephyr-elf/bin/../lib/gcc/aarch64-zephyr-elf/10.2.0"  -L/home/carlo/work/proj/intel/zephyr/zephyr-master/build/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -mcpu=cortex-a53  -mabi=lp64  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -no-pie  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn && cd /home/carlo/work/proj/intel/zephyr/zephyr-master/build/zephyr && /usr/bin/cmake -E echo
/home/carlo/work/proj/intel/sdk/aarch64-zephyr-elf/bin/../lib/gcc/aarch64-zephyr-elf/10.2.0/../../../../aarch64-zephyr-elf/bin/ld:zephyr/linker_zephyr_prebuilt.cmd:59: syntax error
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

--- a/include/arch/arm/aarch64/scripts/linker.ld
+++ b/include/arch/arm/aarch64/scripts/linker.ld
@@ -123,7 +123,7 @@ SECTIONS
 #include <linker/kobject-text.ld>
 
         MMU_ALIGN;
-    } GROUP_LINK_IN(ROMABLE_REGION)
+    } GROUP_ROM_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
 
     _image_text_end = .;
     _image_text_size = _image_text_end - _image_text_start;
@@ -135,7 +135,7 @@ SECTIONS
          * .ARM.extab section containing exception unwinding information.
          */
         *(.ARM.extab* .gnu.linkonce.armextab.*)
-    } GROUP_LINK_IN(ROMABLE_REGION)
+    } GROUP_ROM_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
 #endif
 
     SECTION_PROLOGUE(.ARM.exidx,,)
@@ -151,7 +151,7 @@ SECTIONS
         *(.ARM.exidx* gnu.linkonce.armexidx.*)
 #endif
         __exidx_end = .;
-    } GROUP_LINK_IN(ROMABLE_REGION)
+    } GROUP_ROM_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
 
         _image_rodata_start = .;
 
@@ -173,7 +173,7 @@ SECTIONS
 
 #include <linker/kobject-rom.ld>
 
-    } GROUP_LINK_IN(ROMABLE_REGION)
+    } GROUP_ROM_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
 
 #include <linker/cplusplus-rom.ld>
     MMU_ALIGN;
@@ -243,7 +243,7 @@ SECTIONS
          * section ends on a 4 byte boundary. This wastes a maximum of 3 bytes.
                  */
         __bss_end = ALIGN(4);
-    } GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
+    } GROUP_NOLOAD_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
 
 #include <linker/common-noinit.ld>
 
@@ -301,7 +301,7 @@ SECTIONS
     /* Must be last in romable region */
     SECTION_PROLOGUE(.last_section,(NOLOAD),)
     {
-    } GROUP_LINK_IN(ROMABLE_REGION)
+    } GROUP_ROM_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
 
     /* To provide the image size as a const expression,
      * calculate this value here. */

